numpy/core/tests/test_numeric.py

```diff
@@ -2656,6 +2656,49 @@ def test_outer_out_param():
     assert_equal(np.outer(arr2, arr3, out2), out2)
 
 
+class TestIndices(object):
+
+    def test_simple(self):
+        [X, Y] = np.indices((4,3))
+        assert_array_equal(X, np.array([[0, 0, 0],
+                                        [1, 1, 1],
+                                        [2, 2, 2],
+                                        [3, 3, 3]]))
+        assert_array_equal(Y, np.array([[0, 1, 2],
+                                        [0, 1, 2],
+                                        [0, 1, 2],
+                                        [0, 1, 2]]))
+
+    def test_single_input(self):
+        [X] = np.indices((4,))
+        assert_array_equal(X, np.array([0, 1, 2, 3]))
+
+        [X] = np.indices((4,), sparse=True)
+        assert_array_equal(X, np.array([0, 1, 2, 3]))
+
+    def test_scalar_input(self):
+        assert_array_equal([], np.indices(()))
+        assert_array_equal([], np.indices((), sparse=True))
+        assert_array_equal([[]], np.indices((0,)))
+        assert_array_equal([[]], np.indices((0,), sparse=True))
+
+    def test_sparse(self):
+        [X, Y] = np.indices((4,3), sparse=True)
+        assert_array_equal(X, np.array([[0], [1], [2], [3]]))
+        assert_array_equal(Y, np.array([[0, 1, 2]]))
+
+    def test_return_type(self):
```

Comment:
Could use parameterize as well here, which may be a bit nicer. If you do not have a special path for 0D can leave this, otherwise, it would be good to add a test for 0D dtype as well.

Reply:
At first I didn't really get your `parametrize` suggestion, but now I implemented it. And even though the special path for 0D is gone I still added a test for 0D as well.